server/src/routes/course/student.ts

```diff
@@ -95,6 +95,20 @@ export const updateStudent = (_: ILogger) => async (ctx: Router.RouterContext) =
     setResponse(ctx, BAD_REQUEST, null);
     return;
   }
+  const user = ctx.state!.user;
+  const { githubId: userGithubId, isAdmin, isSupervisor, isManager } = user;
```

Comment:
it doesn't look right. `isSupervisor, isManager` are part of course roles/user

Reply:
try to reuse `userGuards` from `server/src/routes/guards.ts` 

```typescript
const guard = userGuards(ctx.state!.user);
const isPowerUser = guard.isPowerUser(courseId);
const isMentor = guard.isPowerUser(courseId);
...
```

Reply:
done